File: 백준/Silver/2164. 카드2/카드2.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int ret;
        Deque<Integer> dq = new ArrayDeque<>();

        for (int i = 1; i <= n; i++) {
            dq.addLast(i);
        }

        while (true) {
            if (dq.size() == 1)
                break;
            dq.removeFirst();
            int now = dq.peekFirst();
            dq.removeFirst();
            dq.addLast(now);
        }

        System.out.println(dq.peekFirst());
        scanner.close();
    }
}

File: 백준/Silver/2164. 카드2/카드2.cc
#include <bits/stdc++.h>
using namespace std;
int n, ret;
deque<int> dq;

int main()
{
    cin >> n;
    for (int i = 1; i <= n; i++)
    {
        dq.push_back(i);
    }

    while (true)
    {
        if (dq.size() == 1)
            break;
        dq.pop_front();
        int now = dq.front();
        dq.pop_front();
        dq.push_back(now);
    }
    cout << dq.front() << "\n";

    return 0;
}